src/app/components/layouts/overlay/terra-overlay.component.spec.ts

```diff
@@ -1,11 +1,298 @@
+
+        expect(onOutputSpy).toHaveBeenCalled();
+    });
+
+    it(`should call 'emitOutputOnShow' after it has been hidden`, () =>
+    {
+        let onOutputSpy:Spy = spyOn(component, 'emitOutputOnShow');
+        divElement.triggerEventHandler('onShown', null);
+
+        expect(onOutputSpy).toHaveBeenCalled();
+    });
+
+    it(`should set modal sizes (classes) according to the inputs`, () =>
+    {
+        expect(modalDialogElement.classes['modal-lg']).toBe(true); // default
+
+        component.inputIsExtraLarge = true;
+        component.inputIsLarge = false;
+        component.inputIsSmall = true;
+
+        fixture.detectChanges();
+
+        expect(modalDialogElement.classes['modal-lg']).toBe(false);
+        expect(modalDialogElement.classes['modal-xl']).toBe(true);
+        expect(modalDialogElement.classes['modal-sm']).toBe(true);
+    });
+
+    it(`should set 'inputIsStatic' depending on 'inputIsClosable'`, () =>
+    {
+        component.inputIsCloseable = false;
+        fixture.detectChanges();
+        component.ngAfterViewInit();
+
+        expect(component.inputIsStatic).toBe(true);
+        expect(component.viewChildOverlay.config.backdrop).toBe('static');
+        expect(component.viewChildOverlay.config.keyboard).toBe(false);
+    });
+
+    it(`should show header depending on 'inputOverlayTitle'`, () =>
+    {
+        component.inputOverlayTitle = overlayTitle;
+        component.inputIsCloseable = false;
+
+        fixture.detectChanges();
+        let header:DebugElement = fixture.debugElement.query(By.css('div.modal-header'));
+
+        expect(header).toBeTruthy(); // should be present because inputOverlayTitle is set
+
+    });
+
+    it(`should show header depending on 'inputOverlayTitle' and 'inputIsClosable'`, () =>
+    {
+        component.inputOverlayTitle = overlayTitle;
+        component.inputIsCloseable = true;
+
+        fixture.detectChanges();
+
+        let header:DebugElement = fixture.debugElement.query(By.css('div.modal-header'));
+
+        expect(header).toBeTruthy(); // should be present because inputOverlayTitle and inputIsCloseable are set
+    });
+
+    it(`should NOT show header depending on 'inputOverlayTitle' or 'inputIsClosable'`, () =>
+    {
+        component.inputOverlayTitle = null;
+        component.inputIsCloseable = false;
+
+        fixture.detectChanges();
+
+        let header:DebugElement = fixture.debugElement.query(By.css('div.modal-header'));
+
+        expect(header).not.toBeTruthy(); // should NOT be present because inputOverlayTitle and inputIsCloseable are NOT set
+    });
+
+    it(`should show close button depending on 'inputIsClosable'`, () =>
+    {
+        component.inputIsCloseable = true;
+        component.inputOverlayTitle = overlayTitle;
+        fixture.detectChanges();
+
+        let header:DebugElement = fixture.debugElement.query(By.css('div.modal-header'));
+
+        expect(header.children.length).toBe(2); // button is present
```

Comment:
Etwas unspezifisch. Vllt eher sowas:
```suggestion
        expect(header.query('terra-button')).toBeTruthy(); // button is present
```